### DISCUSSION TITLE
Enable ajv-errors in Feathersjs Dove V5?

### DISCUSSION THREAD

**Discussion by joezappie** [feathersjs/feathers#2528] [General]
It took a bit of digging to figure out how to enable ajv-errors for the default ajv instance in feathers v5 schema's. To hopefully save someone else 30 minutes, heres how I do it:

1. Create file "ajv.js" with and put the following in it (change to require if needed):

```
// Enable error message for ajv
import ajvErrors from 'ajv-errors';
import { AJV } from '@feathersjs/schema';

AJV.opts.allErrors = true;
ajvErrors(AJV);
```

2. Include that file before you call any schemas (ideally top of app.js or index.js and before services.js is imported):

```
// index.js
import './ajv.js';
```

I was running into an issue with using ESM modules in nodejs where it was automatically moving all the imports before any code. Pasting this in your index.js will not allow it to be called before your services file is imported, in which case ajv will complain about errorMessage in your schemas and abort your application.

**Comment by daffl** [comment]
`schema(definition, ajv)` allows to pass a custom AJV instance. This has been documented now in https://dove.docs.feathersjs.com/api/schema/schema.html#options
